contracts/actions/rari/RariWithdraw.sol

```diff
@@ -0,0 +1,91 @@
+import "../../interfaces/rari/IFundManager.sol";
+
+/// @title Burns rari pool tokens and withdraws stablecoin from pool
+contract RariWithdraw is ActionBase, DSMath {
+    using TokenUtils for address;
+
+    struct Params {
+        address fundManager;
+        address poolTokenAddress;
+        uint256 poolTokensAmountToPull;
+        address from;
+        address stablecoinAddress;
+        uint256 stablecoinAmountToWithdraw;
+        address to;
+    }
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory inputData = parseInputs(_callData);
+
+        inputData.poolTokensAmountToPull = _parseParamUint(
+                    inputData.poolTokensAmountToPull,
+                    _paramMapping[0],
+                    _subData,
+                    _returnValues
+                );
+        inputData.from = _parseParamAddr(inputData.from, _paramMapping[1], _subData, _returnValues);
+
+        inputData.stablecoinAmountToWithdraw = _parseParamUint(
+                    inputData.stablecoinAmountToWithdraw,
+                    _paramMapping[2],
+                    _subData,
+                    _returnValues
+                );
+        inputData.to = _parseParamAddr(inputData.to, _paramMapping[3], _subData, _returnValues);
+
+        uint256 tokensWithdrawn = _rariWithdraw(inputData);
+        return bytes32(tokensWithdrawn);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public payable override {
+        Params memory inputData = parseInputs(_callData);
+        
+        _rariWithdraw(inputData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+    function _rariWithdraw(Params memory _inputData) internal returns (uint256 tokensWithdrawn) {
+        require(_inputData.to != address(0), "Can't send to burn address");
+
+        IFundManager fundManager = IFundManager(_inputData.fundManager);
+
+        // pull tokens if they're not on proxy
+        _inputData.poolTokenAddress.pullTokensIfNeeded(_inputData.from, _inputData.poolTokensAmountToPull);
+        // approve fundManager to burn as many tokens as needed
+        _inputData.poolTokenAddress.approveToken(address(fundManager), type(uint256).max);
+
+        tokensWithdrawn = fundManager.withdraw(IERC20(_inputData.stablecoinAddress).symbol(), _inputData.stablecoinAmountToWithdraw);
+        
+        uint256 poolTokensAfterWithdraw = _inputData.poolTokenAddress.getBalance(address(this));
+        _inputData.poolTokenAddress.withdrawTokens(_inputData.from, poolTokensAfterWithdraw);
+
```

Comment:
Intention here is to return all that's left from rari pool tokens after withdrawal to the address from where we pulled the tokens. Could possibly be a problem in recipes ( in rare cases), but calculating the right amount of dust is hard in this particular case

Reply:
@yippee-ki-yay potrebno misljenje ovde za withdrawal dusta, da li imas druge ideje